### PR TITLE
Unify access to assets needed during rank-setup.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/matching/i_ranking_assets_repo.h [hunk 1]
```diff
@@ -4,15 +4,18 @@
...
 
+namespace search::fef { class OnnxModel; }
 namespace proton::matching {
 
 /**
  * Interface for retrieving a named constant rank value to be used by features in the rank framework.
  * If the given value is not found a nullptr should be returned.
  */
-struct IConstantValueRepo {
+struct IRankingAssetsRepo {
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/matching/i_ranking_assets_repo.h:14]: Please update the interface description.

File: searchcore/src/vespa/searchcore/proton/matching/indexenvironment.cpp [hunk 2]
```diff
@@ -63,18 +63,14 @@ IndexEnvironment::insertField(const search::fef::FieldInfo &field)
...
-                                   OnnxModels onnxModels)
+                                   search::fef::Properties props,
+                                   const IRankingAssetsRepo &constantValueRepo)
   : _tableManager(),
-    _properties(props),
+    _properties(std::move(props)),
     _fieldNames(),
     _fields(),
     _motivation(UNKNOWN),
     _constantValueRepo(constantValueRepo),
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/matching/indexenvironment.cpp:73]: Please update variable name.

File: searchcore/src/vespa/searchcore/proton/matching/indexenvironment.h [hunk 3]
```diff
@@ -26,9 +24,7 @@ class IndexEnvironment : public search::fef::IIndexEnvironment
     FieldNameMap                        _fieldNames;
     std::vector<search::fef::FieldInfo> _fields;
     mutable FeatureMotivation           _motivation;
-    const IConstantValueRepo           &_constantValueRepo;
-    RankingExpressions                  _rankingExpressions;
-    OnnxModels                          _onnxModels;
+    const IRankingAssetsRepo           &_constantValueRepo;
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/matching/indexenvironment.h:27]: Please update variable name.

File: searchcore/src/vespa/searchcore/proton/matching/matcher.cpp [hunk 4]
```diff
@@ -100,10 +100,9 @@ handleGroupingSession(SessionManager &sessionMgr, GroupingContext & groupingCont
 
 }  // namespace proton::matching::<unnamed>
 
-Matcher::Matcher(const search::index::Schema &schema, const Properties &props, const vespalib::Clock &clock,
-                 QueryLimiter &queryLimiter, const IConstantValueRepo &constantValueRepo,
-                 RankingExpressions rankingExpressions, OnnxModels onnxModels, uint32_t distributionKey)
-  : _indexEnv(distributionKey, schema, props, constantValueRepo, std::move(rankingExpressions), std::move(onnxModels)),
+Matcher::Matcher(const search::index::Schema &schema, Properties props, const vespalib::Clock &clock,
+                 QueryLimiter &queryLimiter, const IRankingAssetsRepo &constantValueRepo, uint32_t distributionKey)
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/matching/matcher.cpp:104]: Please update variable name.

File: searchcore/src/vespa/searchcore/proton/server/searchable_doc_subdb_configurer.h [hunk 5]
```diff
@@ -43,7 +43,7 @@ class SearchableDocSubDBConfigurer
     SearchViewHolder            &_searchView;
     FeedViewHolder              &_feedView;
     matching::QueryLimiter      &_queryLimiter;
-    matching::ConstantValueRepo &_constantValueRepo;
+    matching::RankingAssetsRepo &_constantValueRepo;
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/searchable_doc_subdb_configurer.h:46]: Please update variable name.

File: searchcore/src/vespa/searchcore/proton/server/searchable_doc_subdb_configurer.h [hunk 6]
```diff
@@ -69,16 +69,14 @@ class SearchableDocSubDBConfigurer
                                  SearchViewHolder &searchView,
                                  FeedViewHolder &feedView,
                                  matching::QueryLimiter &queryLimiter,
-                                 matching::ConstantValueRepo &constantValueRepo,
+                                 matching::RankingAssetsRepo &constantValueRepo,
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/searchable_doc_subdb_configurer.h:72]: Please update variable name.

File: searchcore/src/vespa/searchcore/proton/server/searchabledocsubdb.h [hunk 7]
```diff
@@ -69,7 +69,7 @@ SearchableDocSubDB : public FastAccessDocSubDB,
     vespalib::VarHolder<SearchableFeedView::SP> _rFeedView;
     vespalib::eval::ConstantTensorLoader        _tensorLoader;
     vespalib::eval::ConstantValueCache          _constantValueCache;
-    matching::ConstantValueRepo                 _constantValueRepo;
+    matching::RankingAssetsRepo                 _constantValueRepo;
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/searchabledocsubdb.h:72]: Please update variable name.

**Review by geirst — APPROVED** [review]
👍  There is also a conflict to be resolved.

File: searchcore/src/vespa/searchcore/proton/matching/i_ranking_assets_repo.h (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/matching/i_ranking_assets_repo.h:14]: Fixed

File: searchcore/src/vespa/searchcore/proton/matching/indexenvironment.cpp (hunk 2 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/matching/indexenvironment.cpp:73]: Fixed

File: searchcore/src/vespa/searchcore/proton/matching/indexenvironment.h (hunk 3 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/matching/indexenvironment.h:27]: Fixed

File: searchcore/src/vespa/searchcore/proton/matching/matcher.cpp (hunk 4 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/matching/matcher.cpp:104]: Fixed

File: searchcore/src/vespa/searchcore/proton/matching/matcher.h [hunk 8]
```diff
@@ -87,11 +84,9 @@ class Matcher
      * @param props ranking configuration
      * @param clock used for timeout handling
      **/
-    Matcher(const search::index::Schema &schema, const Properties &props,
+    Matcher(const search::index::Schema &schema, Properties props,
             const vespalib::Clock &clock, QueryLimiter &queryLimiter,
-            const IConstantValueRepo &constantValueRepo,
-            RankingExpressions rankingExpressions, OnnxModels onnxModels,
-            uint32_t distributionKey);
+            const IRankingAssetsRepo &constantValueRepo, uint32_t distributionKey);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/matching/matcher.h:89]: Fixed

File: searchcore/src/vespa/searchcore/proton/server/matchers.h [hunk 9]
```diff
@@ -22,20 +22,17 @@ class Matchers {
     std::shared_ptr<matching::Matcher> _fallback;
     std::shared_ptr<matching::Matcher> _default;
 public:
-    typedef std::shared_ptr<Matchers> SP;
-    typedef std::unique_ptr<Matchers> UP;
+    using SP = std::shared_ptr<Matchers>;
     Matchers(const vespalib::Clock &clock,
              matching::QueryLimiter &queryLimiter,
-             const matching::IConstantValueRepo &constantValueRepo);
+             const matching::IRankingAssetsRepo &constantValueRepo);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/matchers.h:28]: Fixed

File: searchcore/src/vespa/searchcore/proton/matching/ranking_assets_repo.h [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+#include "i_ranking_assets_repo.h"
+#include "ranking_constants.h"
+#include "onnx_models.h"
+#include "ranking_expressions.h"
+#include <vespa/eval/eval/value_cache/constant_value.h>
+
+namespace proton::matching {
+
+/**
+ * Class that provides access to a configured set of rank constant values.
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/matching/ranking_assets_repo.h:14]: Fixed

File: searchcore/src/vespa/searchcore/proton/server/matchers.cpp [hunk 11]
```diff
@@ -18,10 +18,10 @@ using namespace vespalib::make_string_short;
 
 Matchers::Matchers(const vespalib::Clock &clock,
                    matching::QueryLimiter &queryLimiter,
-                   const matching::IConstantValueRepo &constantValueRepo)
+                   const matching::IRankingAssetsRepo &constantValueRepo)
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/matchers.cpp:21]: Fixed

File: searchcore/src/vespa/searchcore/proton/server/searchable_doc_subdb_configurer.cpp [hunk 12]
```diff
@@ -89,7 +89,7 @@ SearchableDocSubDBConfigurer(const ISummaryManager::SP &summaryMgr,
                              SearchViewHolder &searchView,
                              FeedViewHolder &feedView,
                              matching::QueryLimiter &queryLimiter,
-                             matching::ConstantValueRepo &constantValueRepo,
+                             matching::RankingAssetsRepo &constantValueRepo,
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/searchable_doc_subdb_configurer.cpp:92]: Fixed

File: searchcore/src/vespa/searchcore/proton/server/searchable_doc_subdb_configurer.h (hunk 5 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/searchable_doc_subdb_configurer.h:45]: Fixed

File: searchcore/src/vespa/searchcore/proton/server/searchable_doc_subdb_configurer.h (hunk 6 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/searchable_doc_subdb_configurer.h:71]: Fixed

File: searchcore/src/vespa/searchcore/proton/server/searchabledocsubdb.h (hunk 7 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/searchabledocsubdb.h:72]: Fixed

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**